src/main/java/com/aws/greengrass/clientdevices/auth/infra/NetworkState.java

```diff
@@ -83,15 +88,15 @@ public ConnectionState getConnectionStateFromMqtt() {
     private void emitNetworkUp() {
         if (isRunning()) {
             for (Consumer<ConnectionState> handler : handlers) {
-                handler.accept(ConnectionState.NETWORK_UP);
+                executorService.submit(() -> handler.accept(ConnectionState.NETWORK_UP));
             }
         }
     }
 
     private void emitNetworkDown() {
         if (isRunning()) {
             for (Consumer<ConnectionState> handler : handlers) {
-                handler.accept(ConnectionState.NETWORK_DOWN);
+                executorService.submit(() -> handler.accept(ConnectionState.NETWORK_DOWN));
```

Comment:
cancel future if network state flips again, or something like that? Any concerns for duplicating calls?

Should each callback run in another thread (which is what this is doing), or should it run the loop in the executor? The current implementation will end up most likely starting _n_ threads for _n_ handlers.